### PR TITLE
Closes #296 document_missing_value_s@devel

### LINKED ISSUE #296: Documentation:  `missing_value` and `missing_values` should be added to common arugments
### Please select a category the issue is focused on?

_No response_

### Let us know where something needs a refresh or put your idea here!

The arguments `missing_value` and `missing_values` are used frequently within our functions.  We should list them into our documentation table. 

We should also add a short discussion to distinguish between the singular and plural of the arguments

image

https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#list-of-common-arguments

## Definition of Done
- [x] `missing_value` and `missing_values`  added to the table
- [x]  Short discussion that distinguishes the use of singular and plural agurments.  `missing_value` versus `missing_values` can be used as an example, but this occurs with other arguments, e.g. `new_var` and `new_vars`
## Example of an update
https://github.com/pharmaverse/admiral/issues/2014

### PR DESCRIPTION
**Pull request by sophie-gem**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/devel/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: vignettes/programming_strategy.Rmd [hunk 1]
```diff
@@ -18,259 +21,328 @@ knitr::opts_chunk$set(
...
+| `date`             | Date of an event / interval. Expects a date object.                                                                                                           |
+| `set_values_to`    | List of variable name-value pairs. Use `process_set_values_to()` for processing the value and providing user friendly error messages.                         |
+| `subject_keys`     | Variables to uniquely identify a subject, defaults to `exprs(STUDYID, USUBJID)`. In function formals, use `subject_keys = get_admiral_option("subject_keys")` |
+| `keep_source_vars` | Specifies which variables from the selected observations should be kept. The default of the argument should be `everything()`.                                |
+| `missing_value`    | A value to be input if the data is missing.                                                                                                                   |
+| `missing_values`   | A named list of expressions where the names are variables in the dataset and the values are a value to be input if the data is missing.                       |
```
Comment (zdz2101) [inline vignettes/programming_strategy.Rmd:295]: This is primarily what was changed right? The markdown wrap 72 thing added up top made this a bit difficult to review.

Perhaps for `missing_values` place a small example like below too
 `exprs(BASEC = "MISSING", BASE = -1)`

File: vignettes/programming_strategy.Rmd [hunk 2]
```diff
@@ -5,8 +5,11 @@ output:
     toc: true
 vignette: >
   %\VignetteIndexEntry{Programming Strategy}
-  %\VignetteEngine{knitr::rmarkdown}
   %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  markdown: 
+    wrap: 72
```
Comment (zdz2101) [inline vignettes/programming_strategy.Rmd:12]: I don't mind adding this as it'll clean up the look for developers moving forward, but makes this current PR a little more difficult to review

**Review by zdz2101 — COMMENTED** [review]
Minor stuff

File: vignettes/programming_strategy.Rmd (hunk 2 above)
Comment (bms63) [reply vignettes/programming_strategy.Rmd:12]: I agree.  Sophie can you revert this change and just show the updates for the missing values.  

Perhaps a separate pr for this wrap could be used if folks want this for easier viewing??

Comment (sophie-gem) [reply vignettes/programming_strategy.Rmd:12]: Of course, no problem - didn't realise this had been altered! Apologies

File: vignettes/programming_strategy.Rmd (hunk 1 above)
Comment (sophie-gem) [reply vignettes/programming_strategy.Rmd:295]: Yep - sounds good, will add.

**Comment by sophie-gem** [comment]
Hopefully that fixes all the issues - but please let me know if I need to do anything else.

File: vignettes/programming_strategy.Rmd [hunk 3]
```diff
@@ -18,259 +18,328 @@ knitr::opts_chunk$set(
...
+| `dtc`              | (Partial) date/datetime in ISO 8601 format.                                                                                                                   |
+| `date`             | Date of an event / interval. Expects a date object.                                                                                                           |
+| `set_values_to`    | List of variable name-value pairs. Use `process_set_values_to()` for processing the value and providing user friendly error messages.                         |
+| `subject_keys`     | Variables to uniquely identify a subject, defaults to `exprs(STUDYID, USUBJID)`. In function formals, use `subject_keys = get_admiral_option("subject_keys")` |
+| `keep_source_vars` | Specifies which variables from the selected observations should be kept. The default of the argument should be `everything()`.                                |
+| `missing_value`    | A value to be input if the data is missing.                                                                                                                   |
```
Comment (bms63) [inline vignettes/programming_strategy.Rmd:291]:
```suggestion
| `missing_value`    | A singular value to be inputted if the data is missing.                                                                                                                   |
```

**Review by bms63 — COMMENTED** [review]
Hey Sophie - still hard to understand where the updates were made.  It is just the sections on singular versus plural and missing_values(s) stuff correct?

File: vignettes/programming_strategy.Rmd (hunk 3 above)
Comment (bms63) [reply vignettes/programming_strategy.Rmd:291]: I had to make sure inputted was a real word lol.  Perhaps another word could be used @zdz2101 what do you think?

**Comment by sophie-gem** [comment]
> Hey Sophie - still hard to understand where the updates were made. It is just the sections on singular versus plural and missing_values(s) stuff correct?

Hey, I'm so sorry - where would I look to see what you can see? (I've no idea why it's so complex) Yep, it's literally the paragraph on the singular versus plural and missing_values.

**Comment by sophie-gem** [comment]
> > Hey Sophie - still hard to understand where the updates were made. It is just the sections on singular versus plural and missing_values(s) stuff correct?
> 
> Hey, I'm so sorry - where would I look to see what you can see? (I've no idea why it's so complex) Yep, it's literally the paragraph on the singular versus plural and missing_values.

Would it be worth rejecting this PR and starting a new branch for this?

**Comment by bms63** [comment]
> > > Hey Sophie - still hard to understand where the updates were made. It is just the sections on singular versus plural and missing_values(s) stuff correct?
> > 
> > 
> > Hey, I'm so sorry - where would I look to see what you can see? (I've no idea why it's so complex) Yep, it's literally the paragraph on the singular versus plural and missing_values.
> 
> Would it be worth rejecting this PR and starting a new branch for this?

No, it is all good.

**Comment by sophie-gem** [comment]
> > > Hey Sophie - still hard to understand where the updates were made. It is just the sections on singular versus plural and missing_values(s) stuff correct?
> > 
> > 
> > Hey, I'm so sorry - where would I look to see what you can see? (I've no idea why it's so complex) Yep, it's literally the paragraph on the singular versus plural and missing_values.
> 
> Usually I can use the diff to see what was update, but looks like the entire document was formatted (?), not sure how, and it flags everything as changed. It is all good I found the sections you udpated. Thanks again for doing this!!
> 
> image

It's probably linked to that wrap thing that appeared at the top of the vignette you identified - no ideas what I did...apologies!

**Comment by sophie-gem** [comment]
Right, that 'should' be easier to review now - I reverted the 'wrapping' in the programming_strategy. The .Rprofile was changed only by the `styler::style_pkg()`. I also used 'entered' instead of 'inputted' to see if that reads better. If anything needs altering, let me know.

**Review by bms63 — APPROVED** [review]
Looks great Spohie thanks for seeing this through!!

**Pull request merged**